syntax_and_semantics/annotations/user_defined_annotations.md

```diff
@@ -0,0 +1,173 @@
+# User-defined annotations
+
+Users can define their own annotations using the `annotation` keyword, which works similarly to defining a `class` or `struct`.
+
+```crystal
+annotation MyAnnotation; end
+```
+
+The annotation can then be applied to various items, including:
+* Instance and class methods
+* Instance variables
+* Classes, structs, and modules
+
+```crystal
+annotation MyAnnotation; end
+
+@[MyAnnotation]
+def foo
+  "foo"
+end
+
+@[MyAnnotation]
+class Klass
+end
+
+@[MyAnnotation]
+module MyModule
+end
+```
+
+## Reading
```

Comment:
Using annotations in macros is a very in-depth topic. Fields and applications should definitely come first.